Week3/homework/3-step3.js

```diff
@@ -1,25 +1,43 @@
 'use strict';
 
 // use a 'for' loop
+
 function repeatStringNumTimesWithFor(str, num) {
-  // add your code here
-  return str;
+  let newStr = '';
+  for (let i = 0; i < num; i++) {
+
+    newStr += str;
+  }
+  return newStr;
 }
 
+
 console.log('for', repeatStringNumTimesWithFor('abc', 3));
 
-// use a 'while' loop
+// // use a 'while' loop
+
 function repeatStringNumTimesWithWhile(str, num) {
-  // add your code here
-  return str;
+  //   // add your code here
+  let newString = "";
+  while (num > 0) {
+    newString += str;
+    num--;
+  }
+  return newString;
 }
 
 console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 
 // use a 'do...while' loop
+
 function repeatStringNumTimesWithDoWhile(str, num) {
-  // add your code here
-  return str;
-}
+  //   // add your code here
+  let string = "";
+  if (num <= 0) return string;
 
-console.log('while', repeatStringNumTimesWithDoWhile('abc', 3));
+  do { string += str; num--; }
+  while (num > 0);
+
+  return string;
+}
+console.log('do..while', repeatStringNumTimesWithDoWhile('abc', 3));
```

Comment:
Nice work on this, all perfectly done. 
Interesting that you decided on the last to test that num was greater than 0 and not before.
I would be inclined for readability to put your two statements (string +=..., and num--) on different lines.
But that is a personal choice.
Nice work.